src/map/WorldMap.kt

```diff
@@ -28,11 +30,13 @@ class CoordinateNodeWorldMap(private val nodes: MutableMap<Coordinate, Node>) :
         val newCoordinate = Coordinate(coordinate.x + deltax, coordinate.y + deltay)
         val oldPlace = nodes[coordinate]
         val newPlace = nodes[newCoordinate]
-        if (oldPlace is OpenSpaceNode && newPlace is OpenSpaceNode) {
+        if (newPlace is WallNode) {
+            throw UnsupportedOperationException("Hit wall node.")
```

Comment:
This feels like your own custom Exception type, eg InvalidMoveException

Reply:
agree